platforms/core-configuration/model-core/src/main/java/org/gradle/internal/extensibility/ConventionAwareHelper.java

```diff
@@ -127,15 +127,8 @@ private boolean mapConventionOn(SupportsConvention target, MappedPropertyImpl ma
     public MappedProperty map(String propertyName, final Closure<?> value) {
         return map(propertyName, new MappedPropertyImpl() {
             @Override
-            public Object doGetValue(org.gradle.api.plugins.Convention convention, IConventionAware conventionAwareObject) {
-                switch (value.getMaximumNumberOfParameters()) {
-                    case 0:
-                        return value.call();
-                    case 1:
-                        return value.call(convention);
-                    default:
-                        return value.call(convention, conventionAwareObject);
-                }
+            public Object doGetValue() {
+                return value.call();
```

Comment:
`it` will  always return `null`. If it is a problem, we should see NPEs in smoke tests.